src/plugins/orangehrmAdminPlugin/Api/I18NTranslationAPI.php

```diff
@@ -0,0 +1,170 @@
+ */
+
+namespace OrangeHRM\Admin\Api;
+
+use OrangeHRM\Admin\Dto\I18NTranslationSearchFilterParams;
+use OrangeHRM\Admin\Traits\Service\LocalizationServiceTrait;
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointCollectionResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\ParameterBag;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+
+class I18NTranslationAPI extends Endpoint implements CollectionEndpoint
+{
+    use LocalizationServiceTrait;
+
+    public const PARAMETER_SOURCE_TEXT = 'sourceText';
+    public const PARAMETER_TRANSLATED_TEXT = 'translatedText';
+    public const PARAMETER_GROUP_ID = 'groupId';
+    public const PARAMETER_ONLY_TRANSLATED = 'onlyTranslated';
+
+    public const PARAMETER_RULE_SOURCE_TEXT_MAX_LENGTH = 250;
+
+    /**
+     * @inheritDoc
+     */
+    public function getAll(): EndpointResult
+    {
+        $i18NTranslationSearchFilterParams = new I18NTranslationSearchFilterParams();
+        $this->setSortingAndPaginationParams($i18NTranslationSearchFilterParams);
+
+        $i18NTranslationSearchFilterParams->setSourceText(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_SOURCE_TEXT,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setTranslatedText(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_TRANSLATED_TEXT,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setGroupId(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_GROUP_ID,
+            )
+        );
+
+        $i18NTranslationSearchFilterParams->setOnlyTranslated(
+            $this->getRequestParams()->getBooleanOrNull(
+                RequestParams::PARAM_TYPE_QUERY,
+                self::PARAMETER_ONLY_TRANSLATED,
+            )
+        );
+
+        $languageId = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, 'languageId');
+        $i18NTranslationSearchFilterParams->setLanguageId($languageId);
+
+        $translations = $this->getLocalizationService()->getLocalizationDao()->getNormalizedTranslations(
+            $i18NTranslationSearchFilterParams
+        );
+
+        $languageCount = $this->getLocalizationService()->getLocalizationDao()->getTranslationsCount(
+            $i18NTranslationSearchFilterParams
+        );
+
+        return new EndpointCollectionResult(
+            ArrayModel::class,
+            array_values($translations),
```

Comment:
maybe you can return array_values from `getNormalizedTranslations` function rather than doing here?